models/QtiRunnerInitDataBuilder.php

```diff
@@ -303,4 +314,28 @@ private function getResponseCountsFromState(array $itemState): int
 
         return $responsesCount;
     }
+
+    /**
+     * @throws MissingOutcomeDeclarationMaxScore
+     */
+    private function getMaxScore($outcomes): float
+    {
+        foreach ($outcomes as $outcome) {
+            if ($outcome[self::OUTCOME_DECLARATION_IDENTIFIER] === self::OUTCOME_DECLARATION_MAXSCORE) {
+                return (float) $outcome[self::OUTCOME_DECLARATION_DEFAULT_VALUE];
+            }
+        }
+
+        throw new MissingOutcomeDeclarationMaxScore();
+    }
+
+    /**
+     * @throws DataItemNotValidException
+     */
+    private function validateItemData($itemData): void
+    {
+        if (!isset($itemData['data'], $itemData['data']['attributes'], $itemData['data']['outcomes'])) {
```

Comment:
BTW, not all items have outcomes. It's perfectly acceptable to have an item with responses, but no outcome declarations. Throwing an exception in such a case seems like an overreaction to me.

Let's consider the outcomes to be an empty array instead, and later call the maxScore calculation as `$this->getMaxScore($itemData['data']['outcomes'])`.